### DISCUSSION TITLE
pinctrl specific pin acces

### DISCUSSION THREAD

**Discussion by milansedlak** [zephyrproject-rtos/zephyr#50762] [Q&A] [answered]
Hello,

I have trouble accessing the required pin directly from the pin controller:
<img width="526" alt="image" src="https://user-images.githubusercontent.com/40634415/192814750-ff408fde-3e91-451d-88b4-dc42abca1ab1.png">

I would like to access, for example, the value for the "SPIM_MOSI" pin, which should be 27.
The output should appear as:
`#define SPIM_MOSI_PIN 27`

Based on the documentation, my best approach was:
`#define SPIM_MOSI_PIN DT_PROP_BY_IDX(DT_PINCTRL_0(spi0, 0), psels, 2)`
But it isn't working. I've tried other approaches like
`DT_PROP_BY_IDX(DT_NODELABEL(spi0_default), psels, 2)`
without any success. 

Could you please tell me where the mistake is?

Many thanks.

**Comment by cameled** [comment, marked as answer]
The NRF_PSEL macro definition as below. After DTS has been processed, there no pin number left.
```
FILE: include/zephyr/dt-bindings/pinctrl/nrf-pinctrl.h

#define NRF_PSEL(fun, port, pin)						       \
	((((((port) * 32U) + (pin)) & NRF_PIN_MSK) << NRF_PIN_POS) |		       \
	 ((NRF_FUN_ ## fun & NRF_FUN_MSK) << NRF_FUN_POS))
```

More refer: https://docs.zephyrproject.org/3.1.0/build/cmake/index.html#build-and-configuration-phases

**Reply by milansedlak** [reply]
Hi,
I've overlooked this. Now it makes sense.
Many thanks
